### PR TITLE
Pin mxnet version in response to mx.metric reorg

### PR DESCRIPTION
**Pull request by zhreshold**
FYI mx.metric is replaced by mx.gluon.metric in mxnet 2.0

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1310-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1310/1/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhreshold** [comment]
@dmlc/gluon-nlp-committers

File: gluoncv/__init__.py [hunk 1]
```diff
@@ -8,6 +8,13 @@
 from .utils.version import _require_mxnet_version, _deprecate_python2
 _deprecate_python2()
 _require_mxnet_version('1.4.0')
+try:
+    # pylint: disable=unused-import
+    from mxnet import metric
+except ImportError:
+    from mxnet.gluon import metric
```
Comment (leezu) [inline gluoncv/__init__.py:15]: There are API changes besides the rename: https://github.com/apache/incubator-mxnet/issues/18046

**Comment by zhreshold** [comment]
@leezu What's nlp's solution in terms of the transition from 1.0 to 2.0 at this time?

**Comment by sxjscience** [comment]
@zhreshold Currently, we are improving lots of APIs. The new numpy version will rely on MXNet 2.0 only and will not be compatible to the old version: https://github.com/dmlc/gluon-nlp/pull/1225

**Comment by zhreshold** [comment]
@sxjscience So you will pin the mxnet to be <2.0 for non-numpy gluonnlp?

**Comment by sxjscience** [comment]
@zhreshold Yes, I think it will depend on MXNet 1.7. Also @dmlc/gluon-nlp-committers , please correct me if I'm wrong.

**Comment by zhreshold** [comment]
@dmlc/gluon-cv-committers @dmlc/gluon-cv-reviewer GluonCV hasn't started transition to either numpy or mxnet 2.0 so the choice will be 
- Maintain compatibility 1.x/2.0, good news is that I have tested that current metric usage is not affected by API but only namespace change
- Drop mxnet 2.0 support for gluoncv master
Any opinion?

**Comment by szha** [comment]
the cost of committing to backward compatibility for 2.0 is too high and infeasible. I think proper versioning should be used to freeze gluoncv to 1.x until the transition happens.

**Comment by bryanyzhu** [comment]
Agree with szha@ and sxjscience@, it is easier to just pin the version to 1.x before the transition is fully ready.

**Comment by mli** [comment]
Job PR-1310-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1310/5/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**